### PR TITLE
feat(android): implement custom protocol

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: src/platform_impl/android/ndk_glue.rs [hunk 1]
```diff
@@ -398,6 +418,152 @@ pub unsafe fn eval(_: JNIEnv, _: JClass, _: JObject) {
...
+        "(Ljava/util/Map;)V",
+        &[response_headers.into()],
+      )?;
+
+      // TODO: this is not working :(
+      /*env.call_method(
+        web_resource_response,
+        "setStatusCodeAndReasonPhrase",
+        "(ILjava/lang/String;)V",
+        &[status_code.into(), env.new_string(reason_phrase)?.into()],
```
Comment (wusyong) [inline src/platform_impl/android/ndk_glue.rs]: Perhaps `status_code` needs to be i32?

File: src/platform_impl/android/ndk_glue.rs [hunk 2]
```diff
@@ -398,6 +418,152 @@ pub unsafe fn eval(_: JNIEnv, _: JClass, _: JObject) {
...
+  Ok(*JObject::null())
+}
+
+#[allow(non_snake_case)]
+pub unsafe fn handleRequest(env: JNIEnv, _: JClass, request: JObject) -> jobject {
+  match handle_request(env, request) {
+    Ok(response) => response,
+    Err(_e) => {
+      #[cfg(debug_assertions)]
+      eprintln!("{}", _e);
```
Comment (wusyong) [inline src/platform_impl/android/ndk_glue.rs]: Let's use log::error instead

File: src/platform_impl/android/ndk_glue.rs (hunk 1 above)
Comment (lucasfernog) [reply src/platform_impl/android/ndk_glue.rs]: You're the best, thanks

File: src/platform_impl/android/ndk_glue.rs (hunk 2 above)
Comment (lucasfernog) [reply src/platform_impl/android/ndk_glue.rs]: done

**Review by wusyong — APPROVED** [review]
I think I'll just merge this asap since it's lgtm already.
I'll approve wry's one too.

**Pull request merged**